coderd/features_internal_test.go

```diff
@@ -0,0 +1,36 @@
+package coderd
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/coder/coder/codersdk"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEntitlements(t *testing.T) {
+	t.Parallel()
+	t.Run("GET", func(t *testing.T) {
+		t.Parallel()
+		r := httptest.NewRequest("GET", "https://example.com/api/v2/entitlements", nil)
+		rw := httptest.NewRecorder()
+		entitlements(rw, r)
+		resp := rw.Result()
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+		dec := json.NewDecoder(resp.Body)
+		var result codersdk.Entitlements
+		err := dec.Decode(&result)
+		require.NoError(t, err)
+		assert.False(t, result.HasLicense)
+		assert.Empty(t, result.Warnings)
+		for _, f := range codersdk.AllFeatures {
+			require.Contains(t, result.Features, f)
+			fe := result.Features[f]
+			assert.False(t, fe.Enabled)
+			assert.Equal(t, codersdk.EntitlementNotEntitled, fe.Entitlement)
+		}
+	})
+}
```

Comment:
Doing this works around any middlewares we have that could make this not function.

Could we do this like a normal `coderd` test instead? Then we'd build `codersdk` functions to fetch entitlements, which seems valuable for future API consumers.

Reply:
I could.  That makes the test more complex to write and slower to execute. This _unit test_ is just testing the handler itself returns what we expect.

There is another test case in `coderd_test.go` that hits the endpoint through all the API routing and middlewares, so that's covered too.

I agree a codersdk function is important, but that's in my next PR that will include a CLI command to display these entitlements.